packages/core/src/queue/event-queue.ts

```diff
@@ -49,25 +50,24 @@ export abstract class CoreEventQueue<
     plugin: Plugin,
     instance: CoreAnalytics
   ): Promise<void> {
-    await Promise.resolve(plugin.load(ctx, instance))
-      .then(() => {
-        this.plugins.push(plugin)
+    if (plugin.type === 'destination' && plugin.name !== 'Segment.io') {
+      plugin.load(ctx, instance).catch((err) => {
+        this.failedInitializations.push(plugin.name)
+        this.emit('initialization_failure', plugin)
+        console.warn(plugin.name, err)
+
+        ctx.log('warn', 'Failed to load destination', {
+          plugin: plugin.name,
+          error: err,
+        })
+
+        this.plugins = this.plugins.filter((p) => p.name !== plugin.name)
```

Comment:
This probably works fine in practice, but we might want to do a strict equality check instead. Reason being, each action of an action destination is implemented as a separate plugin, and they share a name (we use the `alternativeNames` array to distinguish them further if I recall correctly.) So that would mean if for some reason one action failed, but another one succeeded, and they belonged to the same destination, we'd end up with both removed if we just check the name.

This is probably not a real issue in practice since I think currently all those actions would share a load method, but the code change to handle this seems straightforward so may be worth the added protection.
```suggestion
        this.plugins = this.plugins.filter((p) => p === plugin)
```